jdaviz/core/freezable_state.py

```diff
@@ -19,6 +22,84 @@ def __setattr__(self, k, v):
 class FreezableProfileViewerState(ProfileViewerState, FreezableState):
     show_uncertainty = DDCProperty(False, docstring='Whether to show data uncertainties')
 
+    def _reset_x_limits(self, *event):
+        # override glue's _reset_x_limits to account for all layers,
+        # not just reference data (_reset_y_limits already does so)
+        # This is essentially copied directly from Glue's
+        # ProfileViewerState._reset_y_limits but modified for x-limits
```

Comment:
Would it make sense to add this upstream, or is it better this lives in jdaviz for now?

Reply:
@astrofrog - just confirming that you'd prefer both the changes to the profile viewer and image viewer to be in jdaviz and not in glue - at least for now - correct?

Reply:
We can always move it upstream later. Maybe open follow up issue in glue so this doesn't get lost in PR comments.